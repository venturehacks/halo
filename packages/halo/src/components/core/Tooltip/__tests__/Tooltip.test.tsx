import { mount, render } from 'enzyme';
import React from 'react';

import { Tooltip } from '../index';

// NOTE: needed this to prevent the following warning:
//
// Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into
// the server renderer's output format. This will lead to a mismatch between the initial,
// non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in
// components that render exclusively on the client.
//
// Stack overflow link: https://stackoverflow.com/q/58070996/
React.useLayoutEffect = React.useEffect;

describe('Tooltip', () => {
  describe('snapshot', () => {
    describe('size', () => {
      test('sm [default]', () => {
        const component = mount(
          <Tooltip content="small tooltip">
            <span>test</span>
          </Tooltip>,
        );

        component.prop('onMouseEnter');
        component.update();

        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = mount(
          <Tooltip content="medium tooltip" size="md">
            <span>test</span>
          </Tooltip>,
        );

        component.prop('onMouseEnter');
        component.update();

        expect(component).toMatchSnapshot();
      });

      test('lg', () => {
        const component = mount(
          <Tooltip content="large tooltip" size="lg">
            <span>test</span>
          </Tooltip>,
        );

        component.prop('onMouseEnter');
        component.update();

        expect(component).toMatchSnapshot();
      });
    });
  });
});
