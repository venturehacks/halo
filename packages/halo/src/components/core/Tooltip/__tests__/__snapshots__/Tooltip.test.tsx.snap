// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip snapshot size lg 1`] = `
<Tooltip
  content="large tooltip"
  size="lg"
>
  <ForwardRef(TippyWrapper)
    arrow={false}
    className="text-md bg-dark-aaaa py-1 px-2 rounded text-white"
    content="large tooltip"
  >
    <Tippy
      arrow={false}
      className="text-md bg-dark-aaaa py-1 px-2 rounded text-white"
      content="large tooltip"
    >
      <span>
        test
      </span>
      <Portal
        containerInfo={
          <div>
            large tooltip
          </div>
        }
      >
        large tooltip
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Tooltip>
`;

exports[`Tooltip snapshot size md 1`] = `
<Tooltip
  content="medium tooltip"
  size="md"
>
  <ForwardRef(TippyWrapper)
    arrow={false}
    className="text-xs bg-dark-aaaa py-1 px-2 rounded text-white"
    content="medium tooltip"
  >
    <Tippy
      arrow={false}
      className="text-xs bg-dark-aaaa py-1 px-2 rounded text-white"
      content="medium tooltip"
    >
      <span>
        test
      </span>
      <Portal
        containerInfo={
          <div>
            medium tooltip
          </div>
        }
      >
        medium tooltip
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Tooltip>
`;

exports[`Tooltip snapshot size sm [default] 1`] = `
<Tooltip
  content="small tooltip"
>
  <ForwardRef(TippyWrapper)
    arrow={false}
    className="text-xs bg-dark-aaaa py-1 px-2 rounded text-white"
    content="small tooltip"
  >
    <Tippy
      arrow={false}
      className="text-xs bg-dark-aaaa py-1 px-2 rounded text-white"
      content="small tooltip"
    >
      <span>
        test
      </span>
      <Portal
        containerInfo={
          <div>
            small tooltip
          </div>
        }
      >
        small tooltip
      </Portal>
    </Tippy>
  </ForwardRef(TippyWrapper)>
</Tooltip>
`;
